lib/hammer_cli_foreman_discovery/discovery.rb

```diff
@@ -129,6 +129,19 @@ class AutoProvisionCommand < HammerCLIForeman::SingleResourceCommand
       action :auto_provision
       command_name "auto-provision"
       desc _("Auto provision a host")
+
+      option '--all', :flag, _("Auto provision all discovered hosts")
+
+      def execute
+        if option_all?
+          resource.call(:auto_provision_all, {})
+          print_message(success_message) if success_message
```

Comment:
Nitpick: maybe the success/failure messages should be plural because this is a bulk action?